service/api/types.go

```diff
@@ -0,0 +1,93 @@
+package api
+
+// DebuggerState represents the current context of the debugger.
+type DebuggerState struct {
+	// Running is true if the debugged process is continuing execution, and is
+	// false if the debugged process is currently suspended.
+	Running bool `json:"running"`
+	// BreakPoint is the current breakpoint at which the debugged process is
+	// suspended, and may be empty if the process is not suspended.
+	BreakPoint BreakPoint `json:"breakPoint,omitempty"`
+	// Thread is the currently selected debugger thread.
+	Thread *Thread `json:"thread,omitempty"`
+}
+
+// BreakPoint addresses a location at which process execution may be
+// suspended.
+type BreakPoint struct {
+	// ID is a unique identifier for the breakpoint.
+	// TODO: this should probably be a UUID or something.
```

Comment:
A UUID makes things like `clear 1` (to clear breakpoint w/ ID 1) more difficult. I think having a monotonically increasing ID is fine.